packages/starlight/user-components/rehype-steps.ts

```diff
@@ -0,0 +1,52 @@
+import { AstroError } from 'astro/errors';
+import type { Element, Root } from 'hast';
+import { rehype } from 'rehype';
+
+const stepsProcessor = rehype()
+	.data('settings', { fragment: true })
+	.use(function steps() {
+		return (tree: Root) => {
+			const rootElements = tree.children.filter((item): item is Element => item.type === 'element');
+			const [rootElement] = rootElements;
+
+			if (!rootElement) {
+				throw new StepsError(
+					'The `<Steps>` component expects its content to be a single ordered list (`<ol>`) but found no child elements.'
+				);
+			} else if (rootElements.length > 1) {
+				throw new StepsError(
+					'The `<FileTree>` component expects its content to be a single ordered list (`<ol>`) but found multiple child elements: ' +
+						rootElements.map((element: Element) => `\`<${element.tagName}>\``).join(', ') +
+						'.'
+				);
+			} else if (rootElement.tagName !== 'ol') {
+				throw new StepsError(
+					'The `<FileTree>` component expects its content to be a single ordered list (`<ol>`) but found the following element: ' +
```

Comment:
```suggestion
					'The `<Steps>` component expects its content to be a single ordered list (`<ol>`) but found multiple child elements: ' +
						rootElements.map((element: Element) => `\`<${element.tagName}>\``).join(', ') +
						'.'
				);
			} else if (rootElement.tagName !== 'ol') {
				throw new StepsError(
					'The `<Steps>` component expects its content to be a single ordered list (`<ol>`) but found the following element: ' +
```

I intend to play more with the component and code but noticed this typo when randomly throwing some markup at it.

Reply:
Damn, I guess you know where I got this error message from now then 🫣